### PR TITLE
update cifar models for 0.3

### PR DESCRIPTION
**Pull request by brettkoonce**
@rxwei this all works, but could be refactored down more.

I am getting this error when I try to use through on my Conv2DBatchNorm Layer, should I be doing something differently?

```
resnet.swift:133:26: error: cannot invoke 'sequenced' with an argument list of type '(through: Conv2DBatchNorm
)'                                                                                                            
        let test = input.sequenced(through: inputLayer)                                                       
                         ^                                                                                    
resnet.swift:133:26: note: overloads for 'sequenced' exist with these partially matching parameter lists: (thr
ough: L1, L2), (through: L1, L2, L3), (through: L1, L2, L3, L4), (through: L1, L2, L3, L4, L5), (through: L1, 
L2, L3, L4, L5, L6)                                                                                           
        let test = input.sequenced(through: inputLayer)                                                       
                         ^                                                                                    

```

### CODE REVIEW

File: CIFAR/models.swift [hunk 1]
```diff
@@ -3,28 +3,18 @@ import TensorFlow
 // Ported from pytorch.org/tutorials/beginner/blitz/cifar10_tutorial.html
 public struct PyTorchModel: Layer {
     var conv1 = Conv2D<Float>(filterShape: (5, 5, 3, 6), activation: relu)
-    var pool = MaxPool2D<Float>(poolSize: (2, 2), strides: (2, 2))
+    var pool1 = MaxPool2D<Float>(poolSize: (2, 2), strides: (2, 2))
```
Comment (brettkoonce) [inline CIFAR/models.swift]: this pool was being reused, made a copy

**Comment by rxwei** [comment]
> I am getting this error when I try to use through on my Conv2DBatchNorm Layer, should I be doing something differently?

The sequential helper isn't necessary if you need to apply one layer. You can just call `applied(to:)`.

**Comment by brettkoonce** [comment]
@rxwei ahh, doh, give me a sec!

File: CIFAR/main.swift [hunk 2]
```diff
@@ -25,7 +25,7 @@ for epoch in 1...100 {
     for batch in trainingShuffled.batched(Int64(batchSize)) {
         let (labels, images) = (batch.label, batch.data)
         let (loss, gradients) = valueWithGradient(at: model) { model -> Tensor<Float> in
-            let logits = model.applied(to: images, in: Context(learningPhase: .training))
```
Comment (rxwei) [inline CIFAR/main.swift]: This also needs a `Context.local.learningPhase = .training` outside the training loop, otherwise batchnorm won't work.

**Comment by brettkoonce** [comment]
@rxwei actually, this is as far as it can go for now.  because each middle layer is using a relu in this model i have I can't use the pass-through trick.  I could rewrite the class/structs themselves (e.g move the relu logic up a class to Conv2DBatchNorm and add a custom layer for the initial layer and then use .sequenced(through: on the middle layers in the basicblocks) but I would have to sit down and go line by line with the original model and make sure that the logic is the same.  can do, but probably should be in a different pass.  I need to rework the padding logic as well, there's some extraneous calls.  i pinky-promise to revisit this!

File: CIFAR/main.swift (hunk 2 above)
Comment (brettkoonce) [reply CIFAR/main.swift]: okay

Comment (brettkoonce) [reply CIFAR/main.swift]: added a line, but commented it out since resnet is commented out by default as well, the other two models don't use batchnorm, so we shouldn't enable it for them?  or we could just enable it for everything? (but then will it get cargo-cult copied around?)

Comment (rxwei) [reply CIFAR/main.swift]: We should enable training mode on top level.

**Comment by rxwei** [comment]
No problem. Thanks!

File: CIFAR/main.swift (hunk 2 above)
Comment (brettkoonce) [reply CIFAR/main.swift]: tweaked!

File: CIFAR/main.swift [hunk 3]
```diff
@@ -1,16 +1,16 @@
 import TensorFlow
 import Python
 PythonLibrary.useVersion(3)
+//needed to use batch norm in ResNet model layers
```
Comment (rxwei) [inline CIFAR/main.swift]: Could you remove the comment and move it a bit closer to the training loop?

Comment (brettkoonce) [reply CIFAR/main.swift]: put it at head of for loop!
